jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2372,7 +2376,92 @@ else if (oldQueryParams == null || oldQueryParams.size() == 0)
     @Override
     public <T extends HttpUpgradeHandler> T upgrade(Class<T> handlerClass) throws IOException, ServletException
     {
-        throw new ServletException("HttpServletRequest.upgrade() not supported in Jetty");
+        try
```

Comment:
should we assert that response status is 101 and ISE if it isn't?

Reply:
Sounds reasonable. Let me add that check.